inginious/agent/docker_agent/__init__.py

```diff
@@ -490,7 +490,7 @@ def close_and_delete(student_container_id=student_container_id_loop):
                             raise Exception("Feedback file is badly formatted.")
                         elif accepted_types[key] == dict:
                             for sub_key, sub_item in item.items():
-                                if not id_checker(sub_key) or isinstance(sub_item, dict):
+                                if not id_checker(sub_key): # or isinstance(sub_item, dict): Why? Very restrictive.
```

Comment:
This protects the key "custom", "problems" and "tests". I agree that "custom" should be able to contain anything, but "problems" and "tests" must contain a precise format, that should be checked here. Please modify the code to something like

```
elif accepted_types[key] == dict and key != "custom": #custom can contain anything
    for sub_key, sub_item in item.items():
        if not id_checker(sub_key) or isinstance(sub_item, dict):
            raise Exception("Feedback file is badly formatted.")
```

Reply:
Thank you, I just updated the code with your suggestion !